src/group_theory/submonoid/operations.lean

```diff
@@ -828,9 +851,11 @@ def of_left_inverse' (f : M →* N) {g : N → M} (h : function.left_inverse g f
   .. f.mrange_restrict }
 
 /-- A `mul_equiv` `φ` between two monoids `M` and `N` induces a `mul_equiv` between
-a submonoid `S ≤ M` and the submonoid `φ(S) ≤ N`. -/
+a submonoid `S ≤ M` and the submonoid `φ(S) ≤ N`.
+See `monoid_hom.submonoid_map` for a variant for `monoid_hom`s. -/
 @[to_additive "An `add_equiv` `φ` between two additive monoids `M` and `N` induces an `add_equiv`
-between a submonoid `S ≤ M` and the submonoid `φ(S) ≤ N`. ", simps]
+between a submonoid `S ≤ M` and the submonoid `φ(S) ≤ N`. See `add_monoid_hom.add_submonoid_map`
+for a variant for `add_monoid_hom`s.", simps]
 def submonoid_equiv_map (e : M ≃* N) (S : submonoid M) : S ≃* S.map e.to_monoid_hom :=
```

Comment:
I think it is worth changing this to
```lean
{ to_fun := λ x, ⟨e x, _⟩,
  inv_fun := λ x, ⟨e.symm x, _⟩, -- we restate this for `simps` to avoid `⇑e.symm.to_equiv x`
  ..e.to_monoid_hom.submonoid_map S,
  ..e.to_equiv.image S }
```
just so that the link between the two defs is actually in the code.